Projects/Contracts/Sources/Journeys/ContractInformation.swift

```diff
@@ -92,37 +90,42 @@ struct ContractInformationView: View {
                         }
                     }
                 }
-                ForEach(contract.coInsured, id: \.self) { coInsured in
-                    VStack(alignment: .leading) {
-                        hText(coInsured.fullName ?? "")
-                        hText(coInsured.SSN ?? "", style: .footnote)
-                            .foregroundColor(hTextColor.secondary)
+                VStack(spacing: 0) {
+                    ForEach(vm.coInsuredRemainingData(contract: contract), id: \.self) { coInsuredd in
+                        CoInsuredField(
+                            coInsured: coInsuredd,
+                            accessoryView: EmptyView()
+                        )
+                    }
+                    ForEach(vm.coInsuredDeletedData(contract: contract), id: \.self) { coInsured in
+                        CoInsuredField(
+                            coInsured: coInsured,
+                            accessoryView: EmptyView(),
+                            includeStatusPill: StatusPillType.deleted,
+                            date: contract.upcomingChangedAgreement?.activeFrom
+                        )
+                    }
+                    ForEach(vm.coInsuredAddedData(contract: contract), id: \.self) { coInsured in
+                        CoInsuredField(
+                            coInsured: coInsured,
+                            accessoryView: EmptyView(),
+                            includeStatusPill: StatusPillType.added,
+                            date: contract.upcomingChangedAgreement?.activeFrom
+                        )
                     }
                 }
+                .padding(.leading, 16)
 
-                /* TODO: CHANGE WHEN REAL DATA */
                 ForEach(0..<nbOfMissingCoInsured, id: \.self) { index in
-                    hRow {
-                        HStack(alignment: .top) {
-                            VStack(alignment: .leading) {
-                                hText(L10n.contractCoinsured)
-                                hText(L10n.contractNoInformation, style: .footnote)
-                                    .foregroundColor(hTextColor.secondary)
-                            }
-                            .frame(maxWidth: .infinity, alignment: .leading)
-                            Spacer()
-                            Image(uiImage: hCoreUIAssets.warningSmall.image)
-                                .foregroundColor(hSignalColor.amberElement)
-                        }
-                    }
+                    addMissingCoInsuredView(contract: contract)
                     if index < nbOfMissingCoInsured - 1 {
                         hRowDivider()
                     }
                 }
             }
             .withoutHorizontalPadding
             hSection {
-                if nbOfMissingCoInsured != 0 {
+                if nbOfMissingCoInsured != 0 && contract.supportsCoInsured && contract.terminationDate == nil {
```

Comment:
I noticed we do 
contract.supportsCoInsured && contract.terminationDate == nil
few times. Maybe it would be smart to extract it as a separated property in contract
e.g. 
```
var isCoinsuredEnabled: Bool {
self.supportsCoInsured && self.terminationDate == nil
}
```

Reply:
yeah we can do that. The reason I did like this now is because we might only use supportCoInsured in the future (I think it's wrong from backend right now) @sladan-hedvig